lonestar/analytics/cpu/k-shortest-paths/SSSP.cpp

```diff
@@ -0,0 +1,452 @@
+ * parallelism. The code is being released under the terms of the 3-Clause BSD
+ * License (a copy is located in LICENSE.txt at the top-level directory).
+ *
+ * Copyright (C) 2018, The University of Texas at Austin. All rights reserved.
+ * UNIVERSITY EXPRESSLY DISCLAIMS ANY AND ALL WARRANTIES CONCERNING THIS
+ * SOFTWARE AND DOCUMENTATION, INCLUDING ANY WARRANTIES OF MERCHANTABILITY,
+ * FITNESS FOR ANY PARTICULAR PURPOSE, NON-INFRINGEMENT AND WARRANTIES OF
+ * PERFORMANCE, AND ANY WARRANTY THAT MIGHT OTHERWISE ARISE FROM COURSE OF
+ * DEALING OR USAGE OF TRADE.  NO WARRANTY IS EITHER EXPRESS OR IMPLIED WITH
+ * RESPECT TO THE USE OF THE SOFTWARE OR DOCUMENTATION. Under no circumstances
+ * shall University be liable for incidental, special, indirect, direct or
+ * consequential damages or loss of profits, interruption of business, or
+ * related expenses which may arise from use of Software or Documentation,
+ * including but not limited to those resulting from defects in Software and/or
+ * Documentation, or loss or inaccuracy of data of any kind.
+ */
+
+#include <iostream>
+#include <map>
+
+#include <katana/analytics/bfs/bfs.h>
+
+#include "Lonestar/BoilerPlate.h"
+#include "Lonestar/K_SSSP.h"
+#include "katana/AtomicHelpers.h"
+#include "katana/analytics/bfs/bfs_internal.h"
+
+using namespace katana::analytics;
+
+namespace cll = llvm::cl;
+
+static const char* name = "Single Source k Shortest Paths";
+static const char* desc =
+    "Computes the k shortest paths from a source node to all nodes in a "
+    "directed "
+    "graph using a modified chaotic iteration algorithm";
+static const char* url = "k_shortest_paths";
+
+static cll::opt<std::string> inputFile(
+    cll::Positional, cll::desc("<input file>"), cll::Required);
+static cll::opt<unsigned int> startNode(
+    "startNode", cll::desc("Node to start search from (default value 0)"),
+    cll::init(0));
+static cll::opt<unsigned int> reportNode(
+    "reportNode", cll::desc("Node to report distance to (default value 1)"),
+    cll::init(0));
+static cll::opt<unsigned int> stepShift(
+    "delta", cll::desc("Shift value for the deltastep (default value 13)"),
+    cll::init(13));
+static cll::opt<unsigned int> numPaths(
+    "numPaths",
+    cll::desc("Number of paths to compute from source to report node (default "
+              "value 1)"),
+    cll::init(1));
+enum AlgoSSSP { deltaTile = 0, deltaStep, deltaStepBarrier };
+
+const char* const ALGO_NAMES_SSSP[] = {
+    "deltaTile", "deltaStep", "deltaStepBarrier"};
+
+static cll::opt<AlgoSSSP> algoSSSP(
+    "algoSSSP", cll::desc("Choose an algorithm:"),
+    cll::values(
+        clEnumVal(deltaTile, "deltaTile"), clEnumVal(deltaStep, "deltaStep"),
+        clEnumVal(deltaStepBarrier, "deltaStepBarrier")),
+    cll::init(deltaTile));
+
+enum AlgoBFS { asyncBFS = 0, syncBFS };
+
+static cll::opt<AlgoBFS> algoBFS(
+    "algoBFS", cll::desc("Choose an algorithm:"),
```

Comment:
-> "Choose an algorithm for reachability"

BFS and SSSP is confusing. Use "reachability" instead of bfs.